cleanlab/internal/multiannotator_utils.py

```diff
@@ -134,3 +150,122 @@ def format_multiannotator_labels(labels: LabelLike) -> Tuple[pd.DataFrame, dict]
     formatted_labels = labels.replace(label_map)
 
     return formatted_labels, inverse_map
+
+
+def check_consensus_label_classes(
+    labels_multiannotator: pd.DataFrame,
+    consensus_label: np.ndarray,
+    consensus_method: str,
+) -> None:
+    """Check if any classes no longer appear in the set of consensus labels (established using the consensus_method stated)"""
+    unique_ma_labels = np.unique(labels_multiannotator.replace({pd.NA: np.NaN}).astype(float))
+    unique_ma_labels = unique_ma_labels[~np.isnan(unique_ma_labels)]
+    labels_set_difference = set(unique_ma_labels) - set(consensus_label)
+
+    if len(labels_set_difference) > 0:
+        print(
+            "CAUTION: Number of unique classes has been reduced from the original data when establishing consensus labels"
+            f"using consensus method '{consensus_method}', likely due to some classes being rarely annotated."
+            "If training a classifier on these consensus labels, it will never see any of the omitted classes unless you"
+            "manually replace some of the consensus labels."
+            f"Classes in the original data but not in consensus labels: {list(map(int, labels_set_difference))}"
+        )
+
+
+def compute_soft_cross_entropy(
+    labels_multiannotator: pd.DataFrame,
+    pred_probs: np.ndarray,
+) -> float:
+    """Compute soft cross entropy between the annotators' empirical label distribution and model pred_probs"""
+    num_classes = get_num_classes(pred_probs=pred_probs)
+
+    empirical_label_distribution = np.full((len(labels_multiannotator), num_classes), np.NaN)
+    for i in range(len(labels_multiannotator)):
+        s = labels_multiannotator.iloc[i]
+        empirical_label_distribution[i, :] = value_counts(
+            s.dropna(), num_classes=num_classes
+        ) / len(s.dropna())
+
+    clipped_pred_probs = np.clip(pred_probs, a_min=1e-6, a_max=None)
+    soft_cross_entropy = -np.sum(
+        empirical_label_distribution * np.log(clipped_pred_probs), axis=1
+    ) / np.log(num_classes)
+
+    return soft_cross_entropy
+
+
```

Comment:
Another thing to consider to allow user to exert even greater control of runtime would be whether the `find_best_temp_scaler()` search can be conducted on a random subsample of the data and pred_probs.  One probably doesn't need to use all the data to get close to the best temperature value, eg. a maximum sized subsample of num_classes*1e4 or 1e5 probably suffices.

Reply:
There was an inefficiency in the previous iteration of my code - we wanted to scale the labeled and unlabeled pred probs by the same temp. However because the temp scaling for the labeled pred_probs was done within `get_label_quality_multiannotator`, I was not able to get the optimal temperature and had to do the grid search a second time (on the labeled pred_probs) to scale the unlabeled pred_probs.
I've changed that now and so `temp_scale_pred_probs` is not always called right after `find_best_temp_scaler` since the optimal temp is found earlier in the code. 
In this situation making it one `calibrate_pred_probs()` function is a little confusing because I am using the same optimal temp for two sets of pred_probs, so did not implement that yet. But I agree it can be helpful and we can add that function in the future.

Reply:
Other notes:
- added optional args that allows specifying `course_search_range` and `fine_search_size`
- TODO: add a maximum sized subsample to run the grid search on

Reply:
Ah ok sounds good to leave as you said. Just remember this for a future PR to add `calibrate_pred_probs()`, both for multi-annotator and regular classification!  

That future PR could even also consider extending `calibrate_pred_probs()` for multi-label classification, for which the optimal approach is to treat the pred_probs for each class separately as a binary task (one-vs-rest, since they don't need to sum to one across classes) and then apply [sklearn.isotonic](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.isotonic)) as it is utilized inside of sklearn's [CalibratedClassifierCV](https://scikit-learn.org/stable/modules/generated/sklearn.calibration.CalibratedClassifierCV.html#sklearn.calibration.CalibratedClassifierCV)